nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/AbstractHeartbeatMonitor.java

```diff
@@ -343,4 +374,25 @@ public void onLocalNodeIdentifierSet(final NodeIdentifier localNodeId) {
         public void onNodeStateChange(final NodeIdentifier nodeId, final NodeConnectionState newState) {
         }
     }
+
+    private class RemoveHeartbeatsOnNodeRemoval implements ClusterTopologyEventListener {
```

Comment:
@markap14 I'm confused about the difference between `RemoveHeartbeatsOnNodeRemoval` and `ClusterChangeEventListener` which both seem to just remove heartbeats for `onNodeRemoved()`. Is there a reason to not register/unregister `ClusterChangeEventListener` on `start()` and `stop()` and just have one of them?

Reply:
Wow. There is no difference. Somehow I completely missed that we already had that registered. I was expecting to see one registered on start/unregistered on stop and it wasn't there so i added that in to ensure that we were handling the case. Completely missed the fact a listener had already been registered. Great catch. Will remove that.